### PR TITLE
support `local_hub_repo` source in load_forecasts()

### PR DESCRIPTION
**Pull request by Serena-Wang**
- close issue #57 
- create get_forecast_file_path() for files in local hub repo
- refactor load_forecasts() and update documentation

### CODE REVIEW

File: R/load_forecasts_zoltar.R [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                                      query_type = "forecasts",
+                                      units = locations, 
+                                      timezeros = valid_forecast_dates,
+                                      models = models,
+                                      targets = targets,
+                                      types = types,
+                                      verbose = verbose,
+                                      as_of = as_of)
+  if (nrow(forecasts) == 0){
+    warning("Warning in do_zotar_query: Forecasts are not available.\n Please check your parameters.")
```
Comment (elray1) [inline R/load_forecasts_zoltar.R:72]:
```suggestion
    warning("Warning in do_zoltar_query: Forecasts are not available.\n Please check your parameters.")
```

File: R/load_forecasts_repo.R [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#' Load all available forecasts submitted on forecast_dates from local hub repo.
+#' 
+#' The function will throw an error when no forecasts are submitted on 
+#' any dates in forecast_dates for selected models, locations, types and target.
+#'
```
Comment (elray1) [inline R/load_forecasts_repo.R:5]: Add documentation for file_path parameter
```suggestion
#' 
#' @param file_path path to the data-processed folder within a local clone of the hub repo
```

File: R/load_forecasts_zoltar.R [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    # convert value column to double and select columns
+    forecasts <- forecasts %>%
+      dplyr::mutate(value = as.double(value)) %>%
+      tidyr::separate(target, into=c("horizon","temporal_resolution","ahead",
+                                     "target_variable"),
+                      remove = FALSE, extra = "merge") %>%
+      dplyr::rename(location = unit, forecast_date = timezero, type = class) %>%
+      dplyr::select(model, forecast_date, location, horizon, temporal_resolution,
+                    target_variable, type, quantile, value)
```
Comment (elray1) [inline R/load_forecasts_zoltar.R:81]: How much of this is necessary if we got no rows back?  Can we clarify and simplify?

File: R/load_latest_forecasts_repo.R [hunk 4]
```diff
@@ -72,34 +76,54 @@ load_latest_forecasts_repo <- function(file_path, models = NULL, forecast_dates,
...
         paste0(model, "/", forecast_dates, "-", model, ".csv"))
       results_path <- results_path[file.exists(results_path)]
-      results_path <- tail(results_path, 1)
+      
+      if (latest){
+        results_path <- tail(results_path, 1)
+      }
       
       if (length(results_path) == 0) {
+        warning("Warning in get_forcast_file_path: No forecast files are submitted with the given parameters.")
```
Comment (elray1) [inline R/load_latest_forecasts_repo.R:118]:
```suggestion
        warning("Warning in get_forecast_file_path: No forecast files are submitted with the given parameters.")
```

File: R/load_latest_forecasts_repo.R [hunk 5]
```diff
@@ -127,6 +151,10 @@ load_forecast_files_repo <- function(file_paths,
                                      types = NULL,
                                      targets = NULL) {
   # validate file_paths exist
+  if (is.null(file_paths)){
```
Comment (elray1) [inline R/load_latest_forecasts_repo.R:154]: should we also check for `missing(file_paths)` here and/or provide a default of `NULL`?

**Review by elray1 — CHANGES_REQUESTED** [review]
A few minor comments -- thanks!!

File: R/load_forecasts_zoltar.R (hunk 3 above)
Comment (Serena-Wang) [reply R/load_forecasts_zoltar.R]: I simplified this part a bit to include fewer columns. The default data format from Zoltar query has some columns that we are not using now. This part is just cleaning up the data frame when the user is trying to bind results from load_latest_forecasts().

**Pull request merged**